inference-engine/src/mkldnn_plugin/nodes/mkldnn_matmul_node.cpp

```diff
@@ -70,103 +106,135 @@ void MKLDNNMatMulNode::getSupportedDescriptors() {
+        return strides;
+    };
 
-    NodeConfig config;
-    config.dynBatchSupport = true;
+    const std::vector<size_t> inStrides0 = getStridesAndDims(inputShapes[0], transposeIn[0]);
+    const std::vector<size_t> inStrides1 = getStridesAndDims(inputShapes[1], transposeIn[1]);
+    const std::vector<size_t> outStrides = getStridesAndDims(outputShapes[0], false);
 
-    auto createDataConfig = [](const std::vector<size_t>& dims, memory::data_type dataType) -> PortConfig {
-        PortConfig dataConfig;
-        dataConfig.inPlace = -1;
-        dataConfig.constant = false;
-        dataConfig.desc = MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(dims, dataType, MKLDNNMemory::GetPlainFormatByRank(dims.size()));
-        return dataConfig;
-    };
+    inDataDesc[0] = make_unique<MKLDNNMemoryDesc>(inDims0, firstInDataType, inStrides0);
+    inDataDesc[1] = make_unique<MKLDNNMemoryDesc>(inDims1, secondInDataType, inStrides1);
+    outDataDesc   = make_unique<MKLDNNMemoryDesc>(outDims, outputDataType, outStrides);
 
-    config.inConfs.push_back(createDataConfig(getParentEdgeAt(0)->getShape().getStaticDims(), inputDataType0));
-    config.inConfs.push_back(createDataConfig(getParentEdgeAt(1)->getShape().getStaticDims(), inputDataType1));
-    config.outConfs.push_back(createDataConfig(getChildEdgeAt(0)->getShape().getStaticDims(), outputDataType));
+    createDescriptor({inDataDesc[0].get(), inDataDesc[1].get()}, {outDataDesc.get()});
+}
+
+void MKLDNNMatMulNode::createDescriptor(const std::vector<const MemoryDesc*>& inputDesc,
+                                        const std::vector<const MemoryDesc*>& outputDesc) {
+    MKLDNNDescriptor desc{
+        std::shared_ptr<matmul::desc>(
+            new matmul::desc(*inDataDesc[0], *inDataDesc[1], *outDataDesc))};
 
-    supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::gemm_any);
+    descs.push_back(desc);
 }
 
-void MKLDNNMatMulNode::initOptimalPrimitiveDescriptor() {
-    auto selected_pd = getSelectedPrimitiveDescriptor();
-    if (selected_pd == nullptr)
-        IE_THROW()  << errorPrefix << " did not set preferable primitive descriptor";
-    auto config = selected_pd->getConfig();
+void MKLDNNMatMulNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
 
-     if (isConfigDefined(config))
-         return;
+    auto attr = initPrimitiveAttr();
+
+    for (auto& desc : descs) {
+        auto itpd = desc.createPrimitiveDescriptorIterator(getEngine(), *attr);
+        while (static_cast<bool>(itpd)) {
+            NodeConfig config;
+            config.dynBatchSupport = true;
+            for (size_t i = 0; i < descInputNumbers(desc); i++) {
+                PortConfig portConfig;
+                portConfig.inPlace = -1;
+                portConfig.constant = false;
+                portConfig.desc = MemoryDescUtils::applyUndefinedOffset(*getSrcMemDesc(itpd, i));
```

Comment:
Using `MemoryDescUtils::applyUndefinedOffset` means that the primitive is ready to read a tensor with any offset and any strides in general case. M.b. it is not very clear from the function name, but it works this way. As far as I can understand, we place descriptors with specially calculated strides in the `MKLDNNMatMulNode::initSupportedPrimitiveDescriptors` section, but here we declare that we are ready to read with any configured strides. M.b. we should set the descriptor with defined strides only?